osu.Game.Tests/Visual/Gameplay/TestScenePlayerScoreSubmission.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Screens;
+using osu.Game.Online.API;
+using osu.Game.Online.Rooms;
+using osu.Game.Online.Solo;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Screens.Ranking;
+
+namespace osu.Game.Tests.Visual.Gameplay
+{
+    public class TestScenePlayerScoreSubmission : OsuPlayerTestScene
+    {
+        protected override bool AllowFail => false;
+
+        private DummyAPIAccess dummyAPI => (DummyAPIAccess)API;
+
+        protected override bool HasCustomSteps => true;
+
+        private bool allowFail;
+
+        protected override TestPlayer CreatePlayer(Ruleset ruleset)
+        {
+            SelectedMods.Value = !allowFail
+                ? new[] { ruleset.GetAllMods().OfType<ModNoFail>().First() }
+                : Array.Empty<Mod>();
```

Comment:
Out of curiosity more than anything, is there a reason for doing this locally and not using `AllowFail` (the property) and its base logic for setting the mods, by having it delegate to `allowFail` (the field)? As in:

```csharp
protected override bool AllowFail => allowFail;
```

Reply:
Originally I was going in a direction of not using `PlayerTestScene`; this is a left-over from that. I can attempt to reuse that existing variable now I think.